src/compute-client/src/controller/replica.rs

```diff
@@ -255,18 +256,26 @@ where
     /// Most `ComputeCommand`s are independent of the target replica, but some
     /// contain replica-specific fields that must be adjusted before sending.
     fn specialize_command(&self, command: &mut ComputeCommand<T>) {
-        if let ComputeCommand::CreateInstance(InstanceConfig { logging }) = command {
-            *logging = self.config.logging.clone();
-        }
-
-        if let ComputeCommand::CreateTimely { config, epoch } = command {
-            *config = TimelyConfig {
-                workers: self.config.location.workers,
-                process: 0,
-                addresses: self.config.location.dataflow_addrs.clone(),
-                arrangement_exert_proportionality: self.config.arrangement_exert_proportionality,
-            };
-            *epoch = self.epoch;
+        match command {
+            ComputeCommand::CreateTimely { config, epoch } => {
+                *config = TimelyConfig {
+                    workers: self.config.location.workers,
+                    process: 0,
+                    addresses: self.config.location.dataflow_addrs.clone(),
+                    arrangement_exert_proportionality: self
+                        .config
+                        .arrangement_exert_proportionality,
+                };
+                *epoch = self.epoch;
+            }
+            ComputeCommand::CreateInstance(InstanceConfig {
+                logging,
+                expiration_offset,
+            }) => {
+                *logging = self.config.logging.clone();
+                *expiration_offset = COMPUTE_REPLICA_EXPIRATION_OFFSET.get(&self.dyncfg);
```

Comment:
It would be safer to read the config when the replica is added, and store it in the `ReplicaConfig`. Otherwise, we'll refresh it every time there is a disconnect (e.g. caused by a network blip), and if the value changed in between, reconciliation will notice and restart the replica, forcing a rehydration.

The drawback is of course that you can't restart a replica to change its expiration offset, you have to drop and recreate it. But that feels like a minor annoyance.

Reply:
Agree, changed.